File: Week3WeekendAPI/app/src/main/java/com/example/week3weekendapi/RepoListActivity.java
package com.example.week3weekendapi;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.os.Bundle;
import com.example.week3weekendapi.model.datasource.remote.HttpUrlConnectionHelper;
import com.example.week3weekendapi.model.repos.Repository;
import java.util.ArrayList;
import java.util.concurrent.ExecutionException;
import butterknife.BindView;
import butterknife.ButterKnife;

public class RepoListActivity extends AppCompatActivity {
    @BindView(R.id.rvRepoList)
    RecyclerView rvRepoList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_repo_list);
        ButterKnife.bind(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        RepoAsyncTask repoAsyncTask = new RepoAsyncTask(HttpUrlConnectionHelper.MY_REPO_LIST);
        try {
            ArrayList<Repository> repoList = repoAsyncTask.execute().get();
            rvRepoList.setLayoutManager(new LinearLayoutManager(this, RecyclerView.VERTICAL, false));
            RecyclerView.Adapter adapter = new RepoListAdapter(repoList);
            rvRepoList.setAdapter(adapter);
        } catch (ExecutionException e) {
            e.printStackTrace();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: Week3WeekendAPI/app/src/main/java/com/example/week3weekendapi/RepoListActivity.kt
package com.example.week3weekendapi

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import android.os.Bundle
import com.example.week3weekendapi.model.datasource.remote.retrofit.GithubService
import com.example.week3weekendapi.model.repos.Repository
import io.reactivex.Observer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_repo_list.*

class RepoListActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_repo_list)
        initRecyclerView()
    }

    fun initRecyclerView(){
        GithubService.createService().getRepos()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe(object : Observer<List<Repository>>{
                    lateinit var repositories: List<Repository>
                    override fun onComplete() { onResult(repositories) }

                    override fun onSubscribe(d: Disposable) {}

                    override fun onNext(t: List<Repository>) { repositories = t}

                    override fun onError(e: Throwable) {}

                })
    }

    fun onResult(repositories: List<Repository>){
        val adapter = RepoListAdapter(repositories)
        rvRepoList.layoutManager = LinearLayoutManager(this)
        rvRepoList.adapter = adapter
    }

}
